Source/WebCore/platform/graphics/ca/PlatformCAAnimation.cpp

```diff
@@ -76,4 +77,75 @@ bool PlatformCAAnimation::isBasicAnimation() const
     return animationType() == Basic || animationType() == Spring;
 }
 
+static constexpr auto transformKeyPath = "transform"_s;
+static constexpr auto opacityKeyPath = "opacity"_s;
+static constexpr auto backgroundColorKeyPath = "backgroundColor"_s;
+static constexpr auto filterKeyPathPrefix = "filters.filter_"_s;
+#if ENABLE(FILTERS_LEVEL_2)
+static constexpr auto backdropFiltersKeyPath = "backdropFilters"_s;
+#endif
+
+String PlatformCAAnimation::makeKeyPath(AnimatedProperty animatedProperty, FilterOperation::Type filterOperationType, int index)
+{
+    switch (animatedProperty) {
+    case AnimatedProperty::Translate:
+    case AnimatedProperty::Scale:
+    case AnimatedProperty::Rotate:
+    case AnimatedProperty::Transform:
+        return transformKeyPath;
+    case AnimatedProperty::Opacity:
+        return opacityKeyPath;
+    case AnimatedProperty::BackgroundColor:
+        return backgroundColorKeyPath;
+    case AnimatedProperty::Filter:
+        return makeString(filterKeyPathPrefix, index, ".", PlatformCAFilters::animatedFilterPropertyName(filterOperationType));
+#if ENABLE(FILTERS_LEVEL_2)
+    case AnimatedProperty::WebkitBackdropFilter:
+        return backdropFiltersKeyPath;
+#endif
+    case AnimatedProperty::Invalid:
+        ASSERT_NOT_REACHED();
+        return emptyString();
+    }
+    ASSERT_NOT_REACHED();
+    return emptyString();
+}
+
+static bool isValidFilterKeyPath(const String& keyPath)
+{
+    if (!keyPath.startsWith(filterKeyPathPrefix))
+        return false;
+
+    size_t underscoreIndex = filterKeyPathPrefix.length();
+    auto dotIndex = keyPath.find('.', underscoreIndex);
+    if (dotIndex == notFound || dotIndex <= underscoreIndex)
+        return false;
+
+    auto indexString = keyPath.substring(underscoreIndex, dotIndex - underscoreIndex);
+    auto parsedIndex = parseInteger<int>(indexString);
```

Comment:
Are negative values allowed? Maybe `unsigned` ?

Reply:
They're not, hence the `*parsedIndex < 0` check on the next line. But using `parseInteger<unsigned>` would be smarter!